### PR TITLE
sage: 9.5 -> 9.6

### PR DESCRIPTION
**Pull request by collares**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by collares** [comment]
Sage 9.6 is out! This time the upgrade was pretty painless.

**Review by omasanori — APPROVED** [review]
Thank you so much!

Many patches goes away, excellent! I have no claims so far, waiting for CI completion.

**Comment by omasanori** [comment]
By the way, do you have any opinions on following the modularization, @collares ? (Sorry, I should have discussed on the Sage 9.5 PR but I missed the timing)

**Comment by collares** [comment]
I'm very interested in the modularization effort and IMHO we should migrate to the new model. I just didn't have the time to start working on it yet, but I plan on doing so in the near future. Let me know if you have plans too!

File: pkgs/applications/science/math/sage/sage.nix [hunk 1]
```diff
@@ -58,7 +49,7 @@ stdenv.mkDerivation rec {
     quicktest = sage-tests.override { longTests = false; timeLimit = 600; }; # as many tests as possible in ~10m
     doc = sagedoc;
     lib = sage-with-env.env.lib;
-    kernelspec = jupyter-kernel-definition;
+    kernelspec = jupyter-kernel-specs.definition.sagemath;
```
Comment (hqurve) [inline pkgs/applications/science/math/sage/sage.nix:52]: Hi, thanks for updating. The function [`jupyter-kernel.create`](https://github.com/NixOS/nixpkgs/blob/e69af2535c82ab9b85eea6dca1ceb435fee7bb2d/pkgs/applications/editors/jupyter/kernel.nix#L29) does not re-expose the definitions passed to it and hence causes the error

```
error: attribute 'definition' missing

       at <truncated path>/nixpkgs/pkgs/applications/science/math/sage/sage.nix:52:18:

           51|     lib = sage-with-env.env.lib;
           52|     kernelspec = jupyter-kernel-specs.definition.sagemath;
             |                  ^
           53|   };

```

**Pull request merged**